psydac/linalg/tests/test_solvers.py

```diff
@@ -110,14 +110,16 @@ def test_solver_tridiagonal(n, p, dtype, solver, verbose=False):
 
     #Create the solvers
     if solver in ['pcg', 'pbicgstab']:
+        base_solver = solver[1:]
         pc = A.diagonal(inverse=True)
-        solv = inverse(A, solver, pc=pc, tol=1e-13, verbose=verbose, recycle=True)
     else:
-        solv = inverse(A, solver, tol=1e-13, verbose=verbose, recycle=True)
+        base_solver = solver
+        pc = None
+    solv = inverse(A, base_solver, pc=pc, tol=1e-13, verbose=verbose, recycle=True)
     solvt = solv.transpose()
     solvh = solv.H
-    solv2 = inverse(A@A, solver, tol=1e-13, verbose=verbose, recycle=True) # Test solver of composition of operators
-
+    solv2 = inverse(A@A, base_solver, pc=None, tol=1e-13, verbose=verbose, recycle=True) # Test solver of composition of operators
+    
```

Comment:
If we add the parameter `use_jacobi_pc` to this test function, this code would simplify considerably:

```suggestion
    # Create the solvers
    pc = A.diagonal(inverse=True) if use_jacobi_pc else None
    solv  = inverse(A, solver, pc=pc, tol=1e-13, verbose=verbose, recycle=True)
    solvt = solv.transpose()
    solvh = solv.H
    solv2 = inverse(A@A, solver, pc=None, tol=1e-13, verbose=verbose, recycle=True) # Test solver of composition of operators

```

Reply:
done in https://github.com/pyccel/psydac/pull/512/commits/93f9faa810abd470e90536750819730f7fb27726